src/libraries/Microsoft.Extensions.Configuration.Binder/tests/ConfigurationBinderTests.cs

```diff
@@ -112,6 +112,87 @@ public class DerivedOptionsWithIConfigurationSection : DerivedOptions
             public IConfigurationSection DerivedSection { get; set; }
         }
 
+        public record RecordTypeOptions(string Color, int Length);
+
+        public record struct RecordStructTypeOptions(string Color, int Length);
```

Comment:
Here's a list of test cases to consider:

- [ ] readonly struct 
- [ ] immutable class with get/init properties
- [ ] immutable class with get only properties and ctor input 
- [ ] records 
- [ ] class that is partially partially immutable

Reply:
> readonly struct
Added

>  immutable class with get/init properties
Added `CanBindSemiImmutableClassWithInit`

> immutable class with get only properties and ctor input
Already there: `CanBindSemiImmutableClass`

>  records
Already there. Added tests for `record struct` and `readonly record struct`

> class that is partially partially immutable
I _think_ what you want is covered by the test `CanBindSemiImmutableClass`